unittest/factory/residual.cpp

```diff
@@ -85,11 +85,11 @@ boost::shared_ptr<crocoddyl::ResidualModelAbstract> ResidualModelFactory::create
       boost::make_shared<pinocchio::GeometryModel>(pinocchio::GeometryModel());
   pinocchio::GeomIndex ig_frame = geometry->addGeometryObject(
       pinocchio::GeometryObject("frame", frame_index, state->get_pinocchio()->frames[frame_index].parent,
-                                boost::shared_ptr<hpp::fcl::CollisionGeometry>(new hpp::fcl::Sphere(0)), frame_SE3));
+                                CollisionGeometryPtr(new hpp::fcl::Sphere(0)), frame_SE3));
```

Comment:
Perhaps, we should use this code with the correct versions of Pinocchio and HPP-FCL. The idea is to still support old versions of both packages.

Reply:
The [current master branch](https://github.com/stack-of-tasks/pinocchio/blob/483713fff9531c0628ce3de6a5578368865976ac/src/multibody/fcl.hpp#L141) of Pinocchio defines the typedef. The shared_ptr type (boost or std) there is decided [from a compile-time definition](https://github.com/stack-of-tasks/pinocchio/blob/483713fff9531c0628ce3de6a5578368865976ac/src/multibody/fcl.hpp#L67) now.

From what I can see it also exists [in tag 2.6.3](https://github.com/stack-of-tasks/pinocchio/blob/5daf5de1c158363e835d322435bab4aa2f9b9b0b/src/multibody/fcl.hpp#L122) which is the minimum required version in the CMakeLists.

Reply:
I have another question. Have you checked that the Python bindings work well?

I know that boost python was not properly binding 'std::shared_ptr'. I also know that boost python developers were planning to sort out this situation.

Reply:
Boost Python works perfectly with both std and boost shared_ptr.

Reply:
This is what we are using in Pinocchio 3x and more.

Reply:
Thanks for the confirmation!